src/main/java/net/greghaines/jesque/worker/Watchdog.java

```diff
@@ -136,8 +166,39 @@ private String getServerName() {
         }
     }
 
-}
 
+    private void recoverFromRedisRestart() {
+        ExecutorService executor
+                = Executors.newSingleThreadExecutor();
+
+        executor.submit(() -> invokeListener());
+    }
+
+    private void invokeListener() {
+        try {
+            redisRestartRecoveryListener.run();
+            doWorkInPool(this.jedisPool, (PoolUtils.PoolWork<Jedis, Void>) jedis -> {
+                jedis.evalsha(watchdogScriptHash.get(), WATCHDOG_SCRIPT_KEYS_NUMBER, serverName, IS_ALIVE, getCurrTime(), isDebug, RecoveryStatus.FINISHED.name(), String.valueOf(LIGHT_KEEPER_PERIOD));
```

Comment:
Call a different script/job. Calling the same exact script & job is weird and not helping to the readability.
We're not notifying is_alive here. We're just updating the status to FINISHED.